From d6edafd91b0d8c4a8873847ec9d8d4686c131288 Mon Sep 17 00:00:00 2001
From: Pradeep Sriram <pradeep.sriram@amlogic.com>
Date: Fri, 27 Dec 2019 17:07:36 -0800
Subject: [PATCH] RDK: add source switch to test app

PD#SWPL-24384

Problem:
  tv server needs source switching

Solution:
  added option to test app

Verify:
  Local

Change-Id: Ie214a79f4cf7fdb455951d89290331cbe7c08306
---
 test/main_tvtest.cpp | 29 ++++++++++++++++++++++++-----
 1 file changed, 24 insertions(+), 5 deletions(-)

diff --git a/test/main_tvtest.cpp b/test/main_tvtest.cpp
index d9e7ae9..9728c5c 100644
--- a/test/main_tvtest.cpp
+++ b/test/main_tvtest.cpp
@@ -22,6 +22,7 @@
 
 class TvTest: public TvClient::TvClientIObserver {
 public:
+    tv_source_input_t CurrentSource;
     TvTest()
     {
         mpTvClient = TvClient::GetInstance();
@@ -32,7 +33,6 @@ public:
     {
 
     }
-
     void onTvClientEvent(CTvEvent &event)
     {
         int eventType = event.getEventType();
@@ -64,9 +64,9 @@ public:
     int SendCmd(const char *data) {
         LOGD("%s: cmd is %s.\n", __FUNCTION__, data);
         if (strcmp(data, "start") == 0) {
-            mpTvClient->StartTv(SOURCE_HDMI1);
+            mpTvClient->StartTv(CurrentSource);
         } else if (strcmp(data, "stop") == 0) {
-            mpTvClient->StopTv(SOURCE_HDMI1);
+            mpTvClient->StopTv(CurrentSource);
         } else {
             LOGE("invalid cmd!\n");
         }
@@ -109,9 +109,10 @@ int main(int argc, char **argv) {
     char Command[1];
     int run = 1;
     DisplayInit();
+    test->CurrentSource=SOURCE_HDMI1;
 
     LOGD("#### please select cmd####\n");
-    LOGD("#### select s to start####\n");
+    LOGD("#### select 1/2/3 to start####\n");
     LOGD("#### select q to stop####\n");
     LOGD("##########################\n");
     while (run) {
@@ -123,10 +124,28 @@ int main(int argc, char **argv) {
             run = 0;
             break;
           }
-          case 's': {
+          case '1': {
+              test->SendCmd("stop");
+              SetOsdBlankStatus("/sys/class/graphics/fb0/blank", 0);
+              test->CurrentSource=SOURCE_HDMI1;
+              test->SendCmd("start");
+              break;
+          }
+         case '2': {
+              test->SendCmd("stop");
+              SetOsdBlankStatus("/sys/class/graphics/fb0/blank", 0);
+              test->CurrentSource=SOURCE_HDMI2;
+              test->SendCmd("start");
+              break;
+          }
+          case '3': {
+              test->SendCmd("stop");
+              SetOsdBlankStatus("/sys/class/graphics/fb0/blank", 0);
+              test->CurrentSource=SOURCE_HDMI3;
               test->SendCmd("start");
               break;
           }
+ 
           default: {
               test->SendCmd("start");
               break;
-- 
2.17.1

